cmd/cluster-kube-apiserver-operator/render/render.go

```diff
@@ -32,65 +53,137 @@ func NewRenderCommand() *cobra.Command {
+	cmd.Flags().StringVar(&renderOpts.assetOutputDir, "asset-output-dir", "", "Output path for rendered manifests.")
+	cmd.Flags().StringVar(&renderOpts.assetInputDir, "asset-input-dir", "", "A path to directory with certificates and secrets.")
+	cmd.Flags().StringVar(&renderOpts.templatesDir, "templates-input-dir", "/usr/share/bootkube/manifests", "A path to a directory with manifest templates.")
+	cmd.Flags().StringVar(&renderOpts.configOverrideFile, "config-override-file", "", "A sparse KubeAPIConfig.kubecontrolplane.config.openshift.io/v1 file (default: kube-apiserver-config-overrides.yaml in the asset-input-dir)")
+	cmd.Flags().StringVar(&renderOpts.configOutputFile, "config-output-file", "", "Output path for the KubeAPIServerConfig yaml file.")
 
 	return cmd
 }
 
 func (r *renderOpts) Validate() error {
+	if len(r.manifest.namespace) == 0 {
+		return errors.New("missing required flag: --manifest-namespace")
+	}
+	if len(r.manifest.image) == 0 {
+		return errors.New("missing required flag: --manifest-image")
+	}
+	if len(r.manifest.imagePullPolicy) == 0 {
+		return errors.New("missing required flag: --manifest-image-pull-policy")
+	}
+	if len(r.manifest.configHostPath) == 0 {
+		return errors.New("missing required flag: --manifest-config-host-path")
+	}
+	if len(r.manifest.configFileName) == 0 {
+		return errors.New("missing required flag: --manifest-config-file-name")
+	}
+	if len(r.manifest.cloudProviderHostPath) == 0 {
+		return errors.New("missing required flag: --manifest-cloud-provider-host-path")
+	}
+
 	if len(r.assetInputDir) == 0 {
 		return errors.New("missing required flag: --asset-output-dir")
 	}
 	if len(r.assetOutputDir) == 0 {
 		return errors.New("missing required flag: --asset-input-dir")
 	}
-	if len(r.manifestTemplatesDir) == 0 {
-		return errors.New("missing required flag: --manifest-templates-dir")
+	if len(r.templatesDir) == 0 {
+		return errors.New("missing required flag: --templates-dir")
+	}
+	if len(r.configOutputFile) == 0 {
+		return errors.New("missing required flag: --config-output-file")
+	}
+
+	return nil
+}
+
+func (r *renderOpts) complete() error {
+	if len(r.configOverrideFile) == 0 {
+		r.configOverrideFile = filepath.Join(r.assetInputDir, "kube-apiserver-config-overrides.yaml")
 	}
+
 	return nil
 }
 
 func (r *renderOpts) Run() error {
-	defaultAssetsConfig := assets.Config{
-		Namespace:           r.namespace,
-		Image:               r.image,
-		ImagePullPolicy:     "IfNotPresent",
-		KubeAPIServerConfig: assets.KubeAPIServerConfig{},
-		ConfigDir:           r.configDir,
-		CloudProviderDir:    r.cloudProviderDir,
+	if err := r.complete(); err != nil {
+		return err
+	}
+
+	assetsConfig := assets.Config{
+		Namespace:             r.manifest.namespace,
+		Image:                 r.manifest.image,
+		ImagePullPolicy:       r.manifest.imagePullPolicy,
+		KubeAPIServerConfig:   assets.KubeAPIServerConfig{},
```

Comment:
why does this need to be initialized here?

Reply:
that's your code, just moved ;-)